src/devices/Bmxx80/Bme680.cs

```diff
@@ -51,49 +58,252 @@ public Bme680(I2cDevice i2cDevice)
 
             _communicationProtocol = CommunicationProtocol.I2c;
             _controlRegister = (byte)Bme680Register.CTRL_MEAS;
+
+            SetDefaultConfiguration();
+        }
+
+        private Sampling _humiditySampling;
+
+        /// <summary>
+        /// Gets or sets the humidity sampling.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when the <see cref="Sampling"/> is set to an undefined mode.</exception>
+        public Sampling HumiditySampling
+        {
+            get => _humiditySampling;
+            set
+            {
+                if (!Enum.IsDefined(typeof(Sampling), value))
+                    throw new ArgumentOutOfRangeException();
+
+                var status = Read8BitsFromRegister((byte)Bme680Register.CTRL_HUM);
+                status = (byte)((status & (byte)~Bme680Mask.HUMIDITY_SAMPLING) | (byte)value);
+
+                _i2cDevice.Write(new[] { (byte)Bme680Register.CTRL_HUM, status });
+                _humiditySampling = value;
+            }
         }
 
+        private Bme680HeaterProfile _currentHeaterProfile;
+
         /// <summary>
-        /// Set the humidity sampling.
+        /// Gets or sets the heater profile to be used for measurements.
+        /// Current heater profile is only set if the chosen profile is configured.
         /// </summary>
-        /// <param name="sampling">The <see cref="Sampling"/> to set.</param>
-        public void SetHumiditySampling(Sampling sampling)
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when the <see cref="Bme680HeaterProfile"/> is set to an undefined profile.</exception>
+        public Bme680HeaterProfile CurrentHeaterProfile
         {
-            var register = (byte)Bme680Register.CTRL_HUM;
-            byte read = Read8BitsFromRegister(register);
+            get => _currentHeaterProfile;
+            set
+            {
+                if (_heaterConfigs.Exists(config => config.HeaterProfile == value))
```

Comment:
One note on this: The Bme680 can save up to 10 heater profiles in its registers. One of these 10 profiles can be set as current profile which will influence the gas resistance measurement. A heater profile needs to be configured by the user before being used to produce meaningful measurements (the registers are just initialized to 0x00).

If an assignment to CurrentHeaterProfile is done with an unconfigured profile, the assignment is ignored and remains set to the old profile. Theoretically we could also switch to an unconfigured profile or configure it automatically. Just wasn't sure what would be expected here.